pkg/daemon/update.go

```diff
@@ -768,9 +768,11 @@ func (dn *Daemon) reboot(rationale string, timeout time.Duration, rebootCmd *exe
 
 	// reboot, executed async via systemd-run so that the reboot command is executed
 	// in the context of the host asynchronously from us
-	err := rebootCmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to reboot")
+	// We're not returning the error from the reboot command as it can be terminated by
+	// the system itself with signal: terminated. We can't catch the subprocess termination signal
+	// either, we just have one for the MCD itself.
+	if err := rebootCmd.Run(); err != nil {
```

Comment:
See also https://github.com/openshift/machine-config-operator/pull/237 - at one point when trying to get rpm-ostree upgrades (with reboots) and Ansible to play nicely together I spent a *lot* of time learning about reboots...it turned out to work a lot better to do basically `systemd-run reboot`.

In that PR I was trying to do something more ambitious but it should work here to just change the reboot command as I did in that commit to e.g.:

```
rebootCmd := exec.Command("systemd-run", "--unit", "machine-config-daemon-reboot",
		"--description", fmt.Sprintf("machine-config-daemon: %s", rationale)
```
or so.

Reply:
I'll change it to that

Reply:
Fixed